### PR TITLE
Consolidate User Model

### PR DESCRIPTION
**Pull request by SergioSanchez12**
- Add new user service
- Move user object to new user service
- Finalize necessary fields for user model
- Create user summary object (for UI) in the lasn service

### CODE REVIEW

File: lasn/v1/lasn.proto [hunk 1]
```diff
@@ -6,10 +6,12 @@ package lasn.v1;
...
   string user_id = 1;
-  string device_id = 2;
-  string name = 3;
-  int64 birthdate = 4;
-  string thumbnail_uri = 5;
+  string name = 2;
+  string birthdate = 3;
+  repeated string photos = 4;
+  repeated string labels = 5;
+  repeated string filters = 6;
```
Comment (lnogueir) [inline lasn/v1/lasn.proto:13]: Make this an appropriate array of filter objects instead of strings.

**Review by lnogueir — COMMENTED** [review]
We'd also need another structure to represent a user nearby, which would be like a subset of the `User`, that one could be called `UserSummary`.

File: lasn/v1/lasn.proto [hunk 2]
```diff
@@ -6,10 +6,12 @@ package lasn.v1;
 
 message UserSummary {
   string user_id = 1;
-  string device_id = 2;
-  string name = 3;
-  int64 birthdate = 4;
-  string thumbnail_uri = 5;
+  string name = 2;
+  string birthdate = 3;
```
[the remark this answers is not in the document]
Comment (SergioSanchez12) [reply lasn/v1/lasn.proto:10]: Initially I had tried to use a datetime but the conversion from proto message to json didn't work. Would need extra code for that edge case. An alternative would be to use int64 and just store a UNIX timestamp.

[the remark this answers is not in the document]
Comment (lnogueir) [reply lasn/v1/lasn.proto:10]: I think that is a great idea. I'm okay with using `int64`.

**Comment by SergioSanchez12** [comment]
> @SergioSanchez12 I was thinking, maybe we also change the name of the service from `LasnService` to `ApproachService`. The original idea was that we'd have a different service for each different thing component, like snap, etc. But it looks like we're gonna keep everything together for the foreseeable future, so might as well name it `ApproachService`.
> 
> EDIT: @SergioSanchez12 actually, I think it would be better if we keep the `LasnService`. However now, we put those functions related to the user model (fetching and creating) in another service, maybe `UserService`. I think that'd be the best practice. That way we don't have one huge service. And then when I create the stories thing, I'd add another one called `StoryService`. This is how I designed this originally.

Yeah I agree that makes more sense

File: lasn/v1/lasn.proto (hunk 1 above)
Comment (SergioSanchez12) [reply lasn/v1/lasn.proto]: Will do in a seperate PR

File: user/v1/user.proto [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+message User {
+  string user_id = 1;
+  string device_id = 2;
+  string name = 3;
+  int64 birthdate = 4;
+  repeated string photos = 5;
+  repeated Label labels = 6;
+  repeated string filters = 7;
+}
```
Comment (lnogueir) [inline user/v1/user.proto:23]: phone number is missing here

File: lasn/v1/lasn.proto [hunk 4]
```diff
@@ -44,18 +32,16 @@ message SubscribeRequest {
 }
 
 message SubscribeResponse {
-  repeated UserSummary user_summaries = 1;
+  repeated user.v1.User users = 1;
```
Comment (lnogueir) [inline lasn/v1/lasn.proto:35]: Here we should be returning a `UserSummary`, not the full user model. We don't want to send another user their filter, service id, phone number etc, or any other user-specific setting that we may add in the future. `UserSummary` should just represent the user as seen by the public, not its personal information.

File: lasn/v1/lasn.proto [hunk 5]
```diff
@@ -4,33 +4,21 @@ option go_package = ".;protos";
...
-message FetchRequest {
+message FetchLasnRequest {
   string user_id = 1;
 }
 
-message FetchResponse {
-  repeated UserSummary user_summaries = 1;
+message FetchLasnResponse {
+  repeated user.v1.User users = 1;
 }
```
Comment (lnogueir) [inline lasn/v1/lasn.proto:22]: I believe we can get rid of these now. We should double check with @jeverd

Comment (SergioSanchez12) [reply lasn/v1/lasn.proto:22]: Good to still have it incase at any point you need the info of the users in your lasn

**Review by lnogueir — APPROVED** [review]
LGTM

**Comment by lnogueir** [comment]
Just update the description of the PR before you merge indicating that you ended up creating a new service (User service).

**Pull request merged**